loki/transformations/tests/test_inline.py

```diff
@@ -367,6 +465,81 @@ def test_inline_member_routines(tmp_path, frontend):
     assert (b == [3., 3., 3.]).all()
 
 
+@pytest.mark.parametrize('frontend', available_frontends(skip={
+    OMNI: "OMNI can't parse the code"}))
```

Comment:
Why? I can't see anything that would usually upset OMNI, and a quick parse with `F_Front` was successful for me.

Reply:
If you execute

```py
from loki import Subroutine, fgen, OMNI, FP
fcode = """
subroutine member_functions
    implicit none
    integer :: i
    real(kind=8) :: a(3)
    contains
    function add_to_a(b, n)
      integer, intent(in) :: n
      real(kind=8), intent(in) :: b(n)
      real(kind=8) :: add_to_a(n)

      do i = 1, n
        add_to_a(i) = a(i) + b(i)
      end do
    end function
end subroutine member_functions
    """
routine = Subroutine.from_source(fcode, frontend=OMNI)
print(fgen(routine))
```

```fortran
SUBROUTINE member_functions ()
  IMPLICIT NONE
  INTEGER :: i
  REAL(KIND=8) :: a(3)
  CONTAINS
  FUNCTION add_to_a (b, n)
    IMPLICIT NONE
    INTEGER, INTENT(IN) :: n
    REAL(KIND=8), INTENT(IN) :: b(n)
    ! HERE: MISSING DIMENSION !!!
    REAL(KIND=8) :: add_to_a
    DO i=1,n
      add_to_a(i) = a(i) + b(i)
    END DO
  END FUNCTION add_to_a
END SUBROUTINE member_functions
```

Same thing for `real(kind=8), dimension(n) :: add_to_a`. This doesn't happen for `FP` and `OFP`.

Reply:
#391 should fix the problem